pkg/goversion/compat.go

```diff
@@ -10,20 +11,30 @@ var (
 	MaxSupportedVersionOfGoMajor = 1
 	MaxSupportedVersionOfGoMinor = 17
 	goTooOldErr                  = fmt.Errorf("Version of Go is too old for this version of Delve (minimum supported version %d.%d, suppress this error with --check-go-version=false)", MinSupportedVersionOfGoMajor, MinSupportedVersionOfGoMinor)
+	goTooOldWarn                 = fmt.Errorf("Warning: version of Go is too old for this version of Delve (minimum supported version %d.%d)", MinSupportedVersionOfGoMajor, MinSupportedVersionOfGoMinor)
 	dlvTooOldErr                 = fmt.Errorf("Version of Delve is too old for this version of Go (maximum supported version %d.%d, suppress this error with --check-go-version=false)", MaxSupportedVersionOfGoMajor, MaxSupportedVersionOfGoMinor)
+	dlvTooOldWarn                = fmt.Errorf("Warning: version of Delve is too old for this version of Go (maximum supported version %d.%d)", MaxSupportedVersionOfGoMajor, MaxSupportedVersionOfGoMinor)
 )
 
 // Compatible checks that the version specified in the producer string is compatible with
 // this version of delve.
-func Compatible(producer string) error {
+func Compatible(producer string, warnonly bool) error {
 	ver := ParseProducer(producer)
 	if ver.IsDevel() {
 		return nil
 	}
 	if !ver.AfterOrEqual(GoVersion{MinSupportedVersionOfGoMajor, MinSupportedVersionOfGoMinor, -1, 0, 0, ""}) {
+		if warnonly {
+			fmt.Fprintf(os.Stderr, "%v\n", goTooOldWarn)
```

Comment:
This should go to logflags.logOut if not nil (same with the other one).

Reply:
There are other warnings that don't use logOut and go to os.Stderr. What makes this one different?
Wouldn't it potentially send the message to a file with log-dest along with other logging? I don't think that's what desirable here.

Reply:
For the most part if we print to stderr we also exit, otherwise it wouldn't be possible to easily distinguish output from delve from the output of the program. I think all the exceptions are in pkg/config/config.go, which should be changed.